From 2f851530f47cdf25fd138c10907340e388532f87 Mon Sep 17 00:00:00 2001
From: William Hua <william.hua@canonical.com>
Date: Sat, 1 Apr 2017 10:04:50 -0400
Subject: [PATCH 26/32] mir: set application name when creating connection

---
 gdk/mir/gdkmirdisplay.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gdk/mir/gdkmirdisplay.c b/gdk/mir/gdkmirdisplay.c
index 570932fabc..8b39e82e3e 100644
--- a/gdk/mir/gdkmirdisplay.c
+++ b/gdk/mir/gdkmirdisplay.c
@@ -125,7 +125,7 @@ _gdk_mir_display_open (const gchar *display_name)
   GdkMirDisplay *display;
   GDBusConnection *session;
 
-  connection = mir_connect_sync (NULL, "GDK-Mir");
+  connection = mir_connect_sync (NULL, g_get_prgname ());
   if (!connection)
      return NULL;
 
-- 
2.11.0

