Index: gtk+-3.0.2/gtk/gtksettings.c
===================================================================
--- gtk+-3.0.2.orig/gtk/gtksettings.c	2011-03-03 12:12:22.000000000 +1100
+++ gtk+-3.0.2/gtk/gtksettings.c	2011-03-08 15:43:51.912921341 +1100
@@ -404,7 +404,7 @@
                                              g_param_spec_string ("gtk-fallback-icon-theme",
                                                                   P_("Fallback Icon Theme Name"),
                                                                   P_("Name of a icon theme to fall back to"),
-                                                                  NULL,
+                                                                  "gnome",
                                                                   GTK_PARAM_READWRITE),
                                              NULL);
   g_assert (result == PROP_FALLBACK_ICON_THEME);
