Index: gtk+3.0-2.91.1/gtk/gtksettings.c
===================================================================
--- gtk+3.0-2.91.1.orig/gtk/gtksettings.c	2010-10-14 04:56:56.000000000 +0200
+++ gtk+3.0-2.91.1/gtk/gtksettings.c	2010-10-18 14:10:13.075672000 +0200
@@ -311,7 +311,7 @@
                                              g_param_spec_string ("gtk-fallback-icon-theme",
 								  P_("Fallback Icon Theme Name"),
 								  P_("Name of a icon theme to fall back to"),
-								  NULL,
+								  "gnome",
 								  GTK_PARAM_READWRITE),
                                              NULL);
   g_assert (result == PROP_FALLBACK_ICON_THEME);
