Index: gtk+3.0-3.3.4/gtk/gtksettings.c
===================================================================
--- gtk+3.0-3.3.4.orig/gtk/gtksettings.c	2011-11-21 14:30:11.000000000 +0100
+++ gtk+3.0-3.3.4/gtk/gtksettings.c	2011-12-07 18:32:58.121620892 +0100
@@ -423,7 +423,7 @@
                                              g_param_spec_string ("gtk-fallback-icon-theme",
                                                                   P_("Fallback Icon Theme Name"),
                                                                   P_("Name of a icon theme to fall back to"),
-                                                                  NULL,
+                                                                  "gnome",
                                                                   GTK_PARAM_READWRITE),
                                              NULL);
   g_assert (result == PROP_FALLBACK_ICON_THEME);
