Index: gtk+-2.91.6/gtk/gtksettings.c
===================================================================
--- gtk+-2.91.6.orig/gtk/gtksettings.c	2010-12-06 02:46:52.000000000 +1100
+++ gtk+-2.91.6/gtk/gtksettings.c	2010-12-07 16:32:56.507103002 +1100
@@ -371,7 +371,7 @@
                                              g_param_spec_string ("gtk-fallback-icon-theme",
 								  P_("Fallback Icon Theme Name"),
 								  P_("Name of a icon theme to fall back to"),
-								  NULL,
+								  "gnome",
 								  GTK_PARAM_READWRITE),
                                              NULL);
   g_assert (result == PROP_FALLBACK_ICON_THEME);
