Index: gtk+3.0-2.91.1/modules/input/imviqr.c
===================================================================
--- gtk+3.0-2.91.1.orig/modules/input/imviqr.c	2010-10-14 04:56:57.000000000 +0200
+++ gtk+3.0-2.91.1/modules/input/imviqr.c	2010-10-18 14:10:13.859672001 +0200
@@ -244,7 +244,7 @@
   N_("Vietnamese (VIQR)"), /* Human readable name */
   GETTEXT_PACKAGE,	   /* Translation domain */
    GTK_LOCALEDIR,	   /* Dir for bindtextdomain (not strictly needed for "gtk+") */
-  "vi"			   /* Languages for which this module is the default */
+  ""			   /* Languages for which this module is the default */
 };
 
 static const GtkIMContextInfo *info_list[] = {
