Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/m17n-db/+bug/191451

Index: gtk+3.0-3.12.2/modules/input/imviqr.c
===================================================================
--- gtk+3.0-3.12.2.orig/modules/input/imviqr.c	2014-06-05 13:04:31.330060360 +0200
+++ gtk+3.0-3.12.2/modules/input/imviqr.c	2014-06-05 13:04:31.326060360 +0200
@@ -242,7 +242,7 @@
   N_("Vietnamese (VIQR)"), /* Human readable name */
   GETTEXT_PACKAGE,	   /* Translation domain */
    GTK_LOCALEDIR,	   /* Dir for bindtextdomain (not strictly needed for "gtk+") */
-  "vi"			   /* Languages for which this module is the default */
+  ""			   /* Languages for which this module is the default */
 };
 
 static const GtkIMContextInfo *info_list[] = {
