Index: gtk+3.0-2.99.3/gtk/gtktreeview.c
===================================================================
--- gtk+3.0-2.99.3.orig/gtk/gtktreeview.c
+++ gtk+3.0-2.99.3/gtk/gtktreeview.c
@@ -10118,7 +10118,8 @@
 
       if (cursor_path)
 	{
-	  if (gtk_tree_selection_get_mode (tree_view->priv->selection) == GTK_SELECTION_MULTIPLE)
+	  if (gtk_tree_selection_get_mode (tree_view->priv->selection) == GTK_SELECTION_MULTIPLE ||
+	      gtk_tree_selection_get_mode (tree_view->priv->selection) == GTK_SELECTION_SINGLE)
 	    gtk_tree_view_real_set_cursor (tree_view, cursor_path, FALSE, FALSE);
 	  else
 	    gtk_tree_view_real_set_cursor (tree_view, cursor_path, TRUE, FALSE);
