Index: gtk+3.0-2.91.1/gtk/gtktreeview.c
===================================================================
--- gtk+3.0-2.91.1.orig/gtk/gtktreeview.c	2010-10-15 03:56:33.000000000 +0200
+++ gtk+3.0-2.91.1/gtk/gtktreeview.c	2010-10-18 14:10:15.415672002 +0200
@@ -9611,7 +9611,8 @@
 
       if (cursor_path)
 	{
-	  if (tree_view->priv->selection->type == GTK_SELECTION_MULTIPLE)
+	  if (tree_view->priv->selection->type == GTK_SELECTION_MULTIPLE ||
+	      tree_view->priv->selection->type == GTK_SELECTION_SINGLE)
 	    gtk_tree_view_real_set_cursor (tree_view, cursor_path, FALSE, FALSE);
 	  else
 	    gtk_tree_view_real_set_cursor (tree_view, cursor_path, TRUE, FALSE);
