Index: gtk+3.0-3.3.4/gtk/updateiconcache.c
===================================================================
--- gtk+3.0-3.3.4.orig/gtk/updateiconcache.c	2011-11-12 03:52:24.000000000 +0100
+++ gtk+3.0-3.3.4/gtk/updateiconcache.c	2011-12-07 18:33:04.593621204 +0100
@@ -660,7 +660,7 @@
 		      directories = g_list_append (directories, g_strdup (subdir));
 		    }
 		  else
-		    dir_index = 0xffff;
+		    continue;
 		}
 		
 	      image = g_new0 (Image, 1);
