diff -Nur gtk+3.0-2.91.0/gtk/gtk.symbols gtk+3.0-2.91.0.ubuntu/gtk/gtk.symbols
--- gtk+3.0-2.91.0/gtk/gtk.symbols	2010-10-13 11:22:12.000000000 +0200
+++ gtk+3.0-2.91.0.ubuntu/gtk/gtk.symbols	2010-10-13 11:23:54.000000000 +0200
@@ -4366,6 +4366,7 @@
 gtk_widget_get_preferred_size
 gtk_widget_get_preferred_width
 gtk_widget_get_preferred_width_for_height
+ubuntu_gtk_widget_set_has_grab
 #endif
 #endif
 
diff -Nur gtk+3.0-2.91.0/gtk/gtkwidget.c gtk+3.0-2.91.0.ubuntu/gtk/gtkwidget.c
--- gtk+3.0-2.91.0/gtk/gtkwidget.c	2010-10-13 11:22:12.000000000 +0200
+++ gtk+3.0-2.91.0.ubuntu/gtk/gtkwidget.c	2010-10-13 11:26:40.000000000 +0200
@@ -12519,3 +12519,14 @@
 {
   widget->priv->height_request_needed = height_request_needed;
 }
+
+void
+ubuntu_gtk_widget_set_has_grab (GtkWidget *widget,
+                                gboolean   has_grab)
+{
+  if (has_grab)
+    GTK_OBJECT_FLAGS (widget) |= GTK_HAS_GRAB;
+  else
+    GTK_OBJECT_FLAGS (widget) &= ~(GTK_HAS_GRAB);
+}
+ 
diff -Nur gtk+3.0-2.91.0/gtk/gtkwidget.h gtk+3.0-2.91.0.ubuntu/gtk/gtkwidget.h
--- gtk+3.0-2.91.0/gtk/gtkwidget.h	2010-09-30 20:29:56.000000000 +0200
+++ gtk+3.0-2.91.0.ubuntu/gtk/gtkwidget.h	2010-10-13 11:26:59.267041015 +0200
@@ -892,6 +892,9 @@
 
 gboolean     gtk_widget_in_destruction (GtkWidget *widget);
 
+void   ubuntu_gtk_widget_set_has_grab  (GtkWidget *widget,
+                                        gboolean   has_grab);
+
 G_END_DECLS
 
 #endif /* __GTK_WIDGET_H__ */
