Index: b/gtk/gtk.symbols
===================================================================
--- a/gtk/gtk.symbols
+++ b/gtk/gtk.symbols
@@ -3956,6 +3956,7 @@
 gtk_wrap_mode_get_type
 ubuntu_gtk_menu_shell_activate_mnemonic
 ubuntu_gtk_menu_shell_activate_first
+ubuntu_gtk_widget_set_has_grab
 ubuntu_menu_proxy_activate_menu
 ubuntu_menu_proxy_get_type
 ubuntu_menu_proxy_get
Index: b/gtk/gtkwidget.c
===================================================================
--- a/gtk/gtkwidget.c
+++ b/gtk/gtkwidget.c
@@ -14072,3 +14072,11 @@
 {
   widget->priv->style = style;
 }
+
+void
+ubuntu_gtk_widget_set_has_grab (GtkWidget *widget,
+                                gboolean   has_grab)
+{
+  _gtk_widget_set_has_grab (widget, has_grab);
+}
+ 
Index: b/gtk/gtkwidget.h
===================================================================
--- a/gtk/gtkwidget.h
+++ b/gtk/gtkwidget.h
@@ -887,6 +887,9 @@
                                                 GdkModifierIntent  intent);
 
 
+void   ubuntu_gtk_widget_set_has_grab  (GtkWidget *widget,
+                                        gboolean   has_grab);
+
 G_END_DECLS
 
 #endif /* __GTK_WIDGET_H__ */
