Index: gtk+-2.91.5/gtk/gtk.symbols
===================================================================
--- gtk+-2.91.5.orig/gtk/gtk.symbols	2010-11-30 11:49:30.000000000 +1100
+++ gtk+-2.91.5/gtk/gtk.symbols	2010-11-30 11:50:09.787861001 +1100
@@ -3445,6 +3445,7 @@
 gtk_window_unstick
 gtk_wrap_mode_get_type G_GNUC_CONST
 ubuntu_gtk_menu_shell_activate_mnemonic
+ubuntu_gtk_widget_set_has_grab
 ubuntu_menu_proxy_get_type G_GNUC_CONST
 ubuntu_menu_proxy_get
 ubuntu_menu_proxy_insert
Index: gtk+-2.91.5/gtk/gtkwidget.c
===================================================================
--- gtk+-2.91.5.orig/gtk/gtkwidget.c	2010-11-30 01:54:15.000000000 +1100
+++ gtk+-2.91.5/gtk/gtkwidget.c	2010-11-30 11:49:50.077861001 +1100
@@ -13191,3 +13191,11 @@
 {
   widget->priv->height_request_needed = height_request_needed;
 }
+
+void
+ubuntu_gtk_widget_set_has_grab (GtkWidget *widget,
+                                gboolean   has_grab)
+{
+  _gtk_widget_set_has_grab (widget, has_grab);
+}
+ 
Index: gtk+-2.91.5/gtk/gtkwidget.h
===================================================================
--- gtk+-2.91.5.orig/gtk/gtkwidget.h	2010-11-24 15:22:51.000000000 +1100
+++ gtk+-2.91.5/gtk/gtkwidget.h	2010-11-30 11:49:50.077861001 +1100
@@ -943,6 +943,9 @@
 
 gboolean     gtk_widget_in_destruction (GtkWidget *widget);
 
+void   ubuntu_gtk_widget_set_has_grab  (GtkWidget *widget,
+                                        gboolean   has_grab);
+
 G_END_DECLS
 
 #endif /* __GTK_WIDGET_H__ */
