Index: gtk+-3.5.2/gtk/gtk.symbols
===================================================================
--- gtk+-3.5.2.orig/gtk/gtk.symbols	2012-05-16 10:09:15.278667039 +1200
+++ gtk+-3.5.2/gtk/gtk.symbols	2012-05-16 10:09:16.550666997 +1200
@@ -3923,6 +3923,7 @@
 gtk_window_unstick
 gtk_wrap_mode_get_type
 ubuntu_gtk_menu_shell_activate_mnemonic
+ubuntu_gtk_widget_set_has_grab
 ubuntu_menu_proxy_get_type
 ubuntu_menu_proxy_get
 ubuntu_menu_proxy_insert
Index: gtk+-3.5.2/gtk/gtkwidget.c
===================================================================
--- gtk+-3.5.2.orig/gtk/gtkwidget.c	2012-05-16 10:09:13.370667104 +1200
+++ gtk+-3.5.2/gtk/gtkwidget.c	2012-05-16 10:09:16.554666997 +1200
@@ -14062,3 +14062,11 @@
 {
   widget->priv->style = style;
 }
+
+void
+ubuntu_gtk_widget_set_has_grab (GtkWidget *widget,
+                                gboolean   has_grab)
+{
+  _gtk_widget_set_has_grab (widget, has_grab);
+}
+ 
Index: gtk+-3.5.2/gtk/gtkwidget.h
===================================================================
--- gtk+-3.5.2.orig/gtk/gtkwidget.h	2012-05-16 10:09:13.370667104 +1200
+++ gtk+-3.5.2/gtk/gtkwidget.h	2012-05-16 10:09:16.554666997 +1200
@@ -887,6 +887,9 @@
                                                 GdkModifierIntent  intent);
 
 
+void   ubuntu_gtk_widget_set_has_grab  (GtkWidget *widget,
+                                        gboolean   has_grab);
+
 G_END_DECLS
 
 #endif /* __GTK_WIDGET_H__ */
