Index: gtk+3.0-2.91.1/gtk/gtk.symbols
===================================================================
--- gtk+3.0-2.91.1.orig/gtk/gtk.symbols	2010-10-18 14:10:16.151672002 +0200
+++ gtk+3.0-2.91.1/gtk/gtk.symbols	2010-10-18 14:10:18.531672000 +0200
@@ -4397,6 +4397,7 @@
 gtk_widget_set_vexpand_set
 gtk_widget_queue_compute_expand
 gtk_widget_compute_expand
+ubuntu_gtk_widget_set_has_grab
 #endif
 #endif
 
Index: gtk+3.0-2.91.1/gtk/gtkwidget.c
===================================================================
--- gtk+3.0-2.91.1.orig/gtk/gtkwidget.c	2010-10-15 03:56:33.000000000 +0200
+++ gtk+3.0-2.91.1/gtk/gtkwidget.c	2010-10-18 14:10:18.547672002 +0200
@@ -13257,3 +13257,11 @@
 {
   widget->priv->height_request_needed = height_request_needed;
 }
+
+void
+ubuntu_gtk_widget_set_has_grab (GtkWidget *widget,
+                                gboolean   has_grab)
+{
+  _gtk_widget_set_has_grab (widget, has_grab);
+}
+ 
Index: gtk+3.0-2.91.1/gtk/gtkwidget.h
===================================================================
--- gtk+3.0-2.91.1.orig/gtk/gtkwidget.h	2010-10-15 22:58:27.000000000 +0200
+++ gtk+3.0-2.91.1/gtk/gtkwidget.h	2010-10-18 14:10:18.547672002 +0200
@@ -913,6 +913,9 @@
 
 gboolean     gtk_widget_in_destruction (GtkWidget *widget);
 
+void   ubuntu_gtk_widget_set_has_grab  (GtkWidget *widget,
+                                        gboolean   has_grab);
+
 G_END_DECLS
 
 #endif /* __GTK_WIDGET_H__ */
