Index: gtk+-2.91.7/gtk/gtk.symbols
===================================================================
--- gtk+-2.91.7.orig/gtk/gtk.symbols	2010-12-22 13:23:11.843372001 +1100
+++ gtk+-2.91.7/gtk/gtk.symbols	2010-12-22 13:23:15.483372000 +1100
@@ -3767,6 +3767,7 @@
 gtk_window_unstick
 gtk_wrap_mode_get_type G_GNUC_CONST
 ubuntu_gtk_menu_shell_activate_mnemonic
+ubuntu_gtk_widget_set_has_grab
 ubuntu_menu_proxy_get_type G_GNUC_CONST
 ubuntu_menu_proxy_get
 ubuntu_menu_proxy_insert
Index: gtk+-2.91.7/gtk/gtkwidget.c
===================================================================
--- gtk+-2.91.7.orig/gtk/gtkwidget.c	2010-12-22 03:36:24.000000000 +1100
+++ gtk+-2.91.7/gtk/gtkwidget.c	2010-12-22 13:23:15.483372000 +1100
@@ -13978,3 +13978,11 @@
 
   return widget->priv->context;
 }
+
+void
+ubuntu_gtk_widget_set_has_grab (GtkWidget *widget,
+                                gboolean   has_grab)
+{
+  _gtk_widget_set_has_grab (widget, has_grab);
+}
+ 
Index: gtk+-2.91.7/gtk/gtkwidget.h
===================================================================
--- gtk+-2.91.7.orig/gtk/gtkwidget.h	2010-12-22 03:36:05.000000000 +1100
+++ gtk+-2.91.7/gtk/gtkwidget.h	2010-12-22 13:23:15.483372000 +1100
@@ -975,6 +975,9 @@
 GtkWidgetPath *   gtk_widget_get_path (GtkWidget *widget);
 
 
+void   ubuntu_gtk_widget_set_has_grab  (GtkWidget *widget,
+                                        gboolean   has_grab);
+
 G_END_DECLS
 
 #endif /* __GTK_WIDGET_H__ */
