Index: gtk+-2.91.6/gtk/gtk.symbols
===================================================================
--- gtk+-2.91.6.orig/gtk/gtk.symbols	2010-12-07 16:33:04.547103002 +1100
+++ gtk+-2.91.6/gtk/gtk.symbols	2010-12-07 16:33:55.087103003 +1100
@@ -3663,6 +3663,7 @@
 gtk_window_unstick
 gtk_wrap_mode_get_type G_GNUC_CONST
 ubuntu_gtk_menu_shell_activate_mnemonic
+ubuntu_gtk_widget_set_has_grab
 ubuntu_menu_proxy_get_type G_GNUC_CONST
 ubuntu_menu_proxy_get
 ubuntu_menu_proxy_insert
Index: gtk+-2.91.6/gtk/gtkwidget.c
===================================================================
--- gtk+-2.91.6.orig/gtk/gtkwidget.c	2010-12-06 14:04:13.000000000 +1100
+++ gtk+-2.91.6/gtk/gtkwidget.c	2010-12-07 16:33:55.097103002 +1100
@@ -13968,3 +13968,11 @@
 
   return widget->priv->context;
 }
+
+void
+ubuntu_gtk_widget_set_has_grab (GtkWidget *widget,
+                                gboolean   has_grab)
+{
+  _gtk_widget_set_has_grab (widget, has_grab);
+}
+ 
Index: gtk+-2.91.6/gtk/gtkwidget.h
===================================================================
--- gtk+-2.91.6.orig/gtk/gtkwidget.h	2010-12-06 02:46:52.000000000 +1100
+++ gtk+-2.91.6/gtk/gtkwidget.h	2010-12-07 16:33:55.097103002 +1100
@@ -978,6 +978,9 @@
 GtkWidgetPath *   gtk_widget_get_path (GtkWidget *widget);
 
 
+void   ubuntu_gtk_widget_set_has_grab  (GtkWidget *widget,
+                                        gboolean   has_grab);
+
 G_END_DECLS
 
 #endif /* __GTK_WIDGET_H__ */
