Index: gtk+3.0/gtk/gtk.symbols
===================================================================
--- gtk+3.0.orig/gtk/gtk.symbols	2012-07-04 09:21:17.000000000 +0200
+++ gtk+3.0/gtk/gtk.symbols	2012-07-04 09:21:51.612191601 +0200
@@ -3941,6 +3941,7 @@
 gtk_wrap_mode_get_type
 ubuntu_gtk_menu_shell_activate_mnemonic
 ubuntu_gtk_menu_shell_activate_first
+ubuntu_gtk_widget_set_has_grab
 ubuntu_menu_proxy_activate_menu
 ubuntu_menu_proxy_get_type
 ubuntu_menu_proxy_get
Index: gtk+3.0/gtk/gtkwidget.c
===================================================================
--- gtk+3.0.orig/gtk/gtkwidget.c	2012-06-25 21:09:25.000000000 +0200
+++ gtk+3.0/gtk/gtkwidget.c	2012-07-04 09:21:20.004190071 +0200
@@ -14063,3 +14063,11 @@
 {
   widget->priv->style = style;
 }
+
+void
+ubuntu_gtk_widget_set_has_grab (GtkWidget *widget,
+                                gboolean   has_grab)
+{
+  _gtk_widget_set_has_grab (widget, has_grab);
+}
+ 
Index: gtk+3.0/gtk/gtkwidget.h
===================================================================
--- gtk+3.0.orig/gtk/gtkwidget.h	2012-06-25 21:09:25.000000000 +0200
+++ gtk+3.0/gtk/gtkwidget.h	2012-07-04 09:21:20.004190071 +0200
@@ -887,6 +887,9 @@
                                                 GdkModifierIntent  intent);
 
 
+void   ubuntu_gtk_widget_set_has_grab  (GtkWidget *widget,
+                                        gboolean   has_grab);
+
 G_END_DECLS
 
 #endif /* __GTK_WIDGET_H__ */
