Index: gtk+-3.6.4/gtk/gtk.symbols
===================================================================
--- gtk+-3.6.4.orig/gtk/gtk.symbols	2013-01-10 16:03:54.753595593 +1300
+++ gtk+-3.6.4/gtk/gtk.symbols	2013-01-10 16:03:54.745595594 +1300
@@ -3972,6 +3972,7 @@
 gtk_wrap_mode_get_type
 ubuntu_gtk_menu_shell_activate_mnemonic
 ubuntu_gtk_menu_shell_activate_first
+ubuntu_gtk_widget_set_has_grab
 ubuntu_menu_proxy_activate_menu
 ubuntu_menu_proxy_get_type
 ubuntu_menu_proxy_get
Index: gtk+-3.6.4/gtk/gtkwidget.c
===================================================================
--- gtk+-3.6.4.orig/gtk/gtkwidget.c	2013-01-10 16:03:54.753595593 +1300
+++ gtk+-3.6.4/gtk/gtkwidget.c	2013-01-10 16:03:54.749595594 +1300
@@ -14139,3 +14139,11 @@
   else
     g_action_muxer_remove (muxer, name);
 }
+
+void
+ubuntu_gtk_widget_set_has_grab (GtkWidget *widget,
+                                gboolean   has_grab)
+{
+  _gtk_widget_set_has_grab (widget, has_grab);
+}
+ 
Index: gtk+-3.6.4/gtk/gtkwidget.h
===================================================================
--- gtk+-3.6.4.orig/gtk/gtkwidget.h	2013-01-10 16:03:54.753595593 +1300
+++ gtk+-3.6.4/gtk/gtkwidget.h	2013-01-10 16:03:54.749595594 +1300
@@ -891,6 +891,9 @@
                                                                          const gchar  *name,
                                                                          GActionGroup *group);
 
+void   ubuntu_gtk_widget_set_has_grab  (GtkWidget *widget,
+                                        gboolean   has_grab);
+
 G_END_DECLS
 
 #endif /* __GTK_WIDGET_H__ */
