Index: gtk+3.0-3.12.2/gtk/gtkeventbox.c
===================================================================
--- gtk+3.0-3.12.2.orig/gtk/gtkeventbox.c	2014-06-05 13:15:49.678082951 +0200
+++ gtk+3.0-3.12.2/gtk/gtkeventbox.c	2014-06-05 13:15:49.674082951 +0200
@@ -411,7 +411,8 @@
                         | GDK_BUTTON_RELEASE_MASK
                         | GDK_EXPOSURE_MASK
                         | GDK_ENTER_NOTIFY_MASK
-                        | GDK_LEAVE_NOTIFY_MASK;
+                        | GDK_LEAVE_NOTIFY_MASK
+                        | GDK_SCROLL_MASK;
 
   priv = GTK_EVENT_BOX (widget)->priv;
 
