Description: Prevent crash when provided a string without a terminating null
Author: Michael Terry <michael.terry@canonical.com>
Bug: https://bugzilla.gnome.org/show_bug.cgi?id=657359
Bug-Ubuntu: https://bugs.launchpad.net/gtk/+bug/831812

Index: gtk+3.0-3.1.16/gtk/gtkcssprovider.c
===================================================================
--- gtk+3.0-3.1.16.orig/gtk/gtkcssprovider.c	2011-09-05 08:52:01.379554654 +0200
+++ gtk+3.0-3.1.16/gtk/gtkcssprovider.c	2011-09-05 08:52:50.189554977 +0200
@@ -2453,6 +2453,11 @@
 
       parse_statement (scanner);
     }
+  else
+    {
+      free_data = g_strndup (data, length);
+      data = free_data;
+    }
 
   gtk_css_scanner_pop_section (scanner, GTK_CSS_SECTION_DOCUMENT);
 }
