Index: gtk+-3.5.6/gtk/gtkicontheme.c
===================================================================
--- gtk+-3.5.6.orig/gtk/gtkicontheme.c	2012-06-15 06:25:15.000000000 +1200
+++ gtk+-3.5.6/gtk/gtkicontheme.c	2012-06-27 17:29:29.282100247 +1200
@@ -1330,23 +1330,6 @@
   
   ensure_valid_themes (icon_theme);
 
-  /* for symbolic icons, do a search in all registered themes first;
-   * a theme that inherits them from a parent theme might provide
-   * an alternative highcolor version, but still expect the symbolic icon
-   * to show up instead.
-   */
-  if (icon_names[0] &&
-      g_str_has_suffix (icon_names[0], "-symbolic"))
-    {
-      for (l = priv->themes; l; l = l->next)
-        {
-          IconTheme *theme = l->data;
-          icon_info = theme_lookup_icon (theme, icon_names[0], size, allow_svg, use_builtin);
-          if (icon_info)
-            goto out;
-        }
-    }
-
   for (l = priv->themes; l; l = l->next)
     {
       IconTheme *theme = l->data;
