Index: gtk+3.0-3.3.18/gtk/gtkmenu.c
===================================================================
--- gtk+3.0-3.3.18.orig/gtk/gtkmenu.c	2012-03-04 03:45:51.000000000 +0100
+++ gtk+3.0-3.3.18/gtk/gtkmenu.c	2012-03-05 23:39:09.570812971 +0100
@@ -1459,7 +1459,6 @@
  * be used instead.
  *
  * Since: 3.0
- * Rename to: gtk_menu_popup
  */
 void
 gtk_menu_popup_for_device (GtkMenu             *menu,
