https://bugzilla.gnome.org/show_bug.cgi?id=737777

diff --git a/gtk/gtkprintbackend.c b/gtk/gtkprintbackend.c
index 31a181b..3dc4037 100644
--- a/gtk/gtkprintbackend.c
+++ b/gtk/gtkprintbackend.c
@@ -756,7 +756,7 @@ request_password (GtkPrintBackend  *backend,
 
   priv->auth_info_required = g_strdupv (ai_required);
   length = g_strv_length (ai_required);
-  priv->auth_info = g_new0 (gchar *, length);
+  priv->auth_info = g_new0 (gchar *, length + 1);
   priv->store_auth_info = FALSE;
 
   dialog = gtk_dialog_new_with_buttons ( _("Authentication"), NULL, GTK_DIALOG_MODAL, 

