From f5db0e7c75e9db6f76f1196fc3d12c8be43911b5 Mon Sep 17 00:00:00 2001
From: William Jon McCann <william.jon.mccann@gmail.com>
Date: Sat, 01 Feb 2014 09:35:54 +0000
Subject: docs: remove spurious para

---
Index: gtk+3.0-3.10.7/gtk/deprecated/gtkactiongroup.c
===================================================================
--- gtk+3.0-3.10.7.orig/gtk/deprecated/gtkactiongroup.c	2014-02-25 11:38:50.647645945 +0100
+++ gtk+3.0-3.10.7/gtk/deprecated/gtkactiongroup.c	2014-02-25 11:38:50.643645945 +0100
@@ -43,7 +43,7 @@
  * actions that act on that document (eg. save, cut/copy/paste, etc). Each
  * window's menus would be constructed from a combination of two action
  * groups.
- * </para>
+ *
  * <para id="Action-Accel">
  * Accelerators are handled by the GTK+ accelerator map. All actions are
  * assigned an accelerator path (which normally has the form
@@ -52,6 +52,7 @@
  * and toolitems take on this accelerator path. The GTK+ accelerator map
  * code makes sure that the correct shortcut is displayed next to the menu
  * item.
+ * </para>
  *
  * <refsect2 id="GtkActionGroup-BUILDER-UI">
  * <title>GtkActionGroup as GtkBuildable</title>
Index: gtk+3.0-3.10.7/gtk/gtkmain.c
===================================================================
--- gtk+3.0-3.10.7.orig/gtk/gtkmain.c	2014-02-25 11:38:50.647645945 +0100
+++ gtk+3.0-3.10.7/gtk/gtkmain.c	2014-02-25 11:38:50.643645945 +0100
@@ -52,7 +52,7 @@
  * example, when an Open button is clicked you might display a
  * #GtkFileChooserDialog. After a callback finishes, GTK+ will return to the
  * main loop and await more user input.
- * </para>
+ *
  * <example>
  * <title>Typical <function>main()</function> function for a GTK+ application</title>
  * <programlisting>
@@ -82,7 +82,7 @@
  * }
  * </programlisting>
  * </example>
- * <para>
+ *
  * It's OK to use the GLib main loop directly instead of gtk_main(), though it
  * involves slightly more typing. See #GMainLoop in the GLib documentation.
  */
