From a4559f1e19d10be801ac6e954d58b34a12ac2d4f Mon Sep 17 00:00:00 2001
From: Benjamin Otte <otte@redhat.com>
Date: Sat, 11 Feb 2012 01:59:19 +0100
Subject: [PATCH 01/12] treeview: Tell accessible about focus column changes

---
 gtk/a11y/gtktreeviewaccessible.c |    7 +++++++
 gtk/a11y/gtktreeviewaccessible.h |    4 ++++
 2 files changed, 11 insertions(+), 0 deletions(-)

diff --git a/gtk/a11y/gtktreeviewaccessible.c b/gtk/a11y/gtktreeviewaccessible.c
index 4fca7fa..adb3a0b 100644
--- a/gtk/a11y/gtktreeviewaccessible.c
+++ b/gtk/a11y/gtktreeviewaccessible.c
@@ -2081,6 +2081,13 @@ _gtk_tree_view_accessible_toggle_visibility (GtkTreeView       *treeview,
 }
 
 void
+_gtk_tree_view_accessible_update_focus_column (GtkTreeView       *treeview,
+                                               GtkTreeViewColumn *old_focus,
+                                               GtkTreeViewColumn *new_focus)
+{
+}
+
+void
 _gtk_tree_view_accessible_add_state (GtkTreeView          *treeview,
                                      GtkRBTree            *tree,
                                      GtkRBNode            *node,
diff --git a/gtk/a11y/gtktreeviewaccessible.h b/gtk/a11y/gtktreeviewaccessible.h
index 58c75e8..7f953b4 100644
--- a/gtk/a11y/gtktreeviewaccessible.h
+++ b/gtk/a11y/gtktreeviewaccessible.h
@@ -74,6 +74,10 @@ void            _gtk_tree_view_accessible_reorder_column(GtkTreeView       *tree
 void            _gtk_tree_view_accessible_toggle_visibility
                                                         (GtkTreeView       *treeview,
                                                          GtkTreeViewColumn *column);
+void            _gtk_tree_view_accessible_update_focus_column
+                                                        (GtkTreeView       *treeview,
+                                                         GtkTreeViewColumn *old_focus,
+                                                         GtkTreeViewColumn *new_focus);
 
 void            _gtk_tree_view_accessible_add_state     (GtkTreeView       *treeview,
                                                          GtkRBTree         *tree,
-- 
1.7.9

