From 4054d531c32d0d28614435fae1d9041b629197bc Mon Sep 17 00:00:00 2001
From: Rico Tzschichholz <ricotz@t-online.de>
Date: Wed, 06 Mar 2013 21:26:21 +0000
Subject: printing: Fix build with -Werror=format-security

---
diff --git a/modules/printbackends/cups/gtkprintbackendcups.c b/modules/printbackends/cups/gtkprintbackendcups.c
index e80821c..4d095de 100644
--- a/modules/printbackends/cups/gtkprintbackendcups.c
+++ b/modules/printbackends/cups/gtkprintbackendcups.c
@@ -637,7 +637,7 @@ gtk_print_backend_cups_print_stream (GtkPrintBackend         *print_backend,
                                                         cups_printer->hostname,
                                                         cups_printer->device_uri,
                                                         GTK_PRINT_BACKEND_CUPS (print_backend)->username);
-          g_snprintf (printer_absolute_uri, HTTP_MAX_URI, cups_printer->printer_uri);
+          g_snprintf (printer_absolute_uri, HTTP_MAX_URI, "%s", cups_printer->printer_uri);
         }
       else
         {
--
cgit v0.9.1
