From 30e07a1f3df53d84e72440bde1a1bdef6db5e5d2 Mon Sep 17 00:00:00 2001
From: Lars Uebernickel <lars.uebernickel@canonical.com>
Date: Wed, 18 Jun 2014 13:13:30 +0200
Subject: [PATCH] gtkpopover: set background and popover css classes

---
 gtk/gtkpopover.c      | 5 +++++
 gtk/gtkstylecontext.h | 9 +++++++++
 2 files changed, 14 insertions(+)

diff --git a/gtk/gtkpopover.c b/gtk/gtkpopover.c
index 6255e0e..6f6f3f0 100644
--- a/gtk/gtkpopover.c
+++ b/gtk/gtkpopover.c
@@ -119,12 +119,17 @@ static void
 gtk_popover_init (GtkPopover *popover)
 {
   GtkWidget *widget;
+  GtkStyleContext *context;
 
   widget = GTK_WIDGET (popover);
   gtk_widget_set_has_window (widget, TRUE);
   popover->priv = gtk_popover_get_instance_private (popover);
   popover->priv->modal = TRUE;
   popover->priv->apply_shape = TRUE;
+
+  context = gtk_widget_get_style_context (widget);
+  gtk_style_context_add_class (context, GTK_STYLE_CLASS_BACKGROUND);
+  gtk_style_context_add_class (context, GTK_STYLE_CLASS_POPOVER);
 }
 
 static void
diff --git a/gtk/gtkstylecontext.h b/gtk/gtkstylecontext.h
index 1bab439..96033b1 100644
--- a/gtk/gtkstylecontext.h
+++ b/gtk/gtkstylecontext.h
@@ -786,6 +786,15 @@ struct _GtkStyleContextClass
  */
 #define GTK_STYLE_CLASS_DESTRUCTIVE_ACTION "destructive-action"
 
+/**
+ * GTK_STYLE_CLASS_POPOVER:
+ *
+ * A CSS class that matches popovers. Used by #GtkPopover.
+ *
+ * Since: 3.14
+ */
+#define GTK_STYLE_CLASS_POPOVER "popover"
+
 /* Predefined set of widget regions */
 
 /**
-- 
2.0.0

