diff -Nur gtk+-3.4.0.orig/gdk/x11/gdkdevicemanager-xi2.c gtk+-3.4.0/gdk/x11/gdkdevicemanager-xi2.c
--- gtk+-3.4.0.orig/gdk/x11/gdkdevicemanager-xi2.c	2012-04-05 11:45:09.995748328 +0200
+++ gtk+-3.4.0/gdk/x11/gdkdevicemanager-xi2.c	2012-04-05 11:45:58.275750665 +0200
@@ -331,9 +331,7 @@
         input_source = GDK_SOURCE_ERASER;
       else if (strstr (tmp_name, "cursor"))
         input_source = GDK_SOURCE_CURSOR;
-      else if (strstr (tmp_name, "finger") ||
-               (strstr (tmp_name, "touch") &&
-                !strstr (tmp_name, "touchpad")))
+      else if (strstr (tmp_name, "finger"))
         input_source = GDK_SOURCE_TOUCHSCREEN;
       else if (strstr (tmp_name, "wacom") ||
                strstr (tmp_name, "pen"))
