Bug-Upstream: https://bugzilla.gnome.org/show_bug.cgi?id=684533

Index: gtk+3.0-3.12.2/modules/printbackends/cups/gtkprintbackendcups.c
===================================================================
--- gtk+3.0-3.12.2.orig/modules/printbackends/cups/gtkprintbackendcups.c	2014-06-05 13:15:49.762082954 +0200
+++ gtk+3.0-3.12.2/modules/printbackends/cups/gtkprintbackendcups.c	2014-06-05 13:15:49.758082954 +0200
@@ -2213,6 +2213,19 @@
 		   &port,
 		   resource, sizeof (resource));
 
+  if (strncmp (method, "dnssd", 5) == 0)
+    {
+      _httpResolveURI(cups_printer->printer_uri,
+		      uri, sizeof(uri), 0, NULL, NULL);
+      httpSeparateURI (HTTP_URI_CODING_ALL,
+		       uri,
+		       method, sizeof (method),
+		       username, sizeof (username),
+		       hostname, sizeof (hostname),
+		       &port,
+		       resource, sizeof (resource));
+    }
+
   if (strncmp (resource, "/printers/", 10) == 0)
     {
       cups_printer->ppd_name = g_strdup (resource + 10);
