diff -Nur gtk+-3.5.8/gtk/gtknotebook.c gtk+-3.5.8.ubuntu/gtk/gtknotebook.c
--- gtk+-3.5.8/gtk/gtknotebook.c	2012-07-16 04:33:24.000000000 +0200
+++ gtk+-3.5.8.ubuntu/gtk/gtknotebook.c	2012-07-18 21:25:18.614189856 +0200
@@ -3150,8 +3150,11 @@
     {
       g_object_ref (page->tab_label);
 
-      if (GTK_IS_WINDOW (parent)) /* parent widget is the drag window */
-        gtk_container_remove (GTK_CONTAINER (parent), page->tab_label);
+      if (GTK_IS_WINDOW (parent))
+        {
+          /* parent widget is the drag window */
+          gtk_container_remove (GTK_CONTAINER (parent), page->tab_label);
+        }
       else
         gtk_widget_unparent (page->tab_label);
 
@@ -4929,17 +4932,7 @@
       page->mnemonic_activate_signal = 0;
 
       gtk_widget_set_state_flags (page->tab_label, 0, TRUE);
-      if (gtk_widget_get_window (page->tab_label) != gtk_widget_get_window (GTK_WIDGET (notebook)) ||
-          !NOTEBOOK_IS_TAB_LABEL_PARENT (notebook, page))
-        {
-          GtkWidget *parent;
-
-          parent = gtk_widget_get_parent (page->tab_label);
-          if (GTK_IS_WINDOW (parent))
-            gtk_container_remove (GTK_CONTAINER (parent), page->tab_label);
-          else
-            gtk_widget_unparent (page->tab_label);
-        }
+      gtk_widget_unparent (page->tab_label);
       page->tab_label = NULL;
     }
 }
