Index: gtk+3.0-3.12.2/po/POTFILES.in
===================================================================
--- gtk+3.0-3.12.2.orig/po/POTFILES.in
+++ gtk+3.0-3.12.2/po/POTFILES.in
@@ -290,27 +290,27 @@ modules/printbackends/file/gtkprintbacke
 modules/printbackends/lpr/gtkprintbackendlpr.c
 modules/printbackends/papi/gtkprintbackendpapi.c
 modules/printbackends/test/gtkprintbackendtest.c
-gtk/resources/ui/gtkaboutdialog.ui.h
-gtk/resources/ui/gtkappchooserdialog.ui.h
-gtk/resources/ui/gtkappchooserwidget.ui.h
-gtk/resources/ui/gtkapplication-quartz.ui.h
-gtk/resources/ui/gtkassistant.ui.h
-gtk/resources/ui/gtkcolorchooserdialog.ui.h
-gtk/resources/ui/gtkcoloreditor.ui.h
-gtk/resources/ui/gtkdialog.ui.h
-gtk/resources/ui/gtkfilechooserbutton.ui.h
-gtk/resources/ui/gtkfilechooserdialog.ui.h
-gtk/resources/ui/gtkfilechooserwidget.ui.h
-gtk/resources/ui/gtkfontbutton.ui.h
-gtk/resources/ui/gtkfontchooserdialog.ui.h
-gtk/resources/ui/gtkfontchooserwidget.ui.h
-gtk/resources/ui/gtkinfobar.ui.h
-gtk/resources/ui/gtklockbutton.ui.h
-gtk/resources/ui/gtkmessagedialog.ui.h
-gtk/resources/ui/gtkpagesetupunixdialog.ui.h
-gtk/resources/ui/gtkpathbar.ui.h
-gtk/resources/ui/gtkprintunixdialog.ui.h
-gtk/resources/ui/gtkrecentchooserdefault.ui.h
-gtk/resources/ui/gtkscalebutton.ui.h
-gtk/resources/ui/gtkstatusbar.ui.h
-gtk/resources/ui/gtkvolumebutton.ui.h
+[type: gettext/glade]gtk/resources/ui/gtkaboutdialog.ui
+[type: gettext/glade]gtk/resources/ui/gtkappchooserdialog.ui
+[type: gettext/glade]gtk/resources/ui/gtkappchooserwidget.ui
+[type: gettext/glade]gtk/resources/ui/gtkapplication-quartz.ui
+[type: gettext/glade]gtk/resources/ui/gtkassistant.ui
+[type: gettext/glade]gtk/resources/ui/gtkcolorchooserdialog.ui
+[type: gettext/glade]gtk/resources/ui/gtkcoloreditor.ui
+[type: gettext/glade]gtk/resources/ui/gtkdialog.ui
+[type: gettext/glade]gtk/resources/ui/gtkfilechooserbutton.ui
+[type: gettext/glade]gtk/resources/ui/gtkfilechooserdialog.ui
+[type: gettext/glade]gtk/resources/ui/gtkfilechooserwidget.ui
+[type: gettext/glade]gtk/resources/ui/gtkfontbutton.ui
+[type: gettext/glade]gtk/resources/ui/gtkfontchooserdialog.ui
+[type: gettext/glade]gtk/resources/ui/gtkfontchooserwidget.ui
+[type: gettext/glade]gtk/resources/ui/gtkinfobar.ui
+[type: gettext/glade]gtk/resources/ui/gtklockbutton.ui
+[type: gettext/glade]gtk/resources/ui/gtkmessagedialog.ui
+[type: gettext/glade]gtk/resources/ui/gtkpagesetupunixdialog.ui
+[type: gettext/glade]gtk/resources/ui/gtkpathbar.ui
+[type: gettext/glade]gtk/resources/ui/gtkprintunixdialog.ui
+[type: gettext/glade]gtk/resources/ui/gtkrecentchooserdefault.ui
+[type: gettext/glade]gtk/resources/ui/gtkscalebutton.ui
+[type: gettext/glade]gtk/resources/ui/gtkstatusbar.ui
+[type: gettext/glade]gtk/resources/ui/gtkvolumebutton.ui
